src/Streamly/Internal/Data/Fold.hs

```diff
@@ -1199,7 +1141,50 @@ spanBy
     -> Fold m a b
     -> Fold m a c
     -> Fold m a (b, c)
-spanBy cmp fld1 fld2 = splitWith (,) (groupBy cmp fld1) fld2
+spanBy cmp (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
```

Comment:
We can make this a parser operation and use the `groupBy` parser combinator to implement it.

Reply:
Let's leave this for now, let's not remove things that are working properly. We can always remove stuff later.

Reply:
Let's leave this for now, let's not remove things that are working properly. We can always remove stuff later.